### PR TITLE
[TS] LPS-118240 Not possible to set a "_score" sort in asc

### PR DESCRIPTION
**Pull request by joshuacords**
[ci:test:search - 57 out of 70 jobs passed in 1 hour 33 minutes
](https://github.com/joshuacords/liferay-portal/pull/93#issuecomment-666876905) (All test failures are known to be unrelated)
[ci:test:stable - 21 out of 21 jobs passed
ci:test:relevant - 40 out of 40 jobs passed in 2 hours 6 minutes](https://github.com/joshuacords/liferay-portal/pull/93#issuecomment-666843094)

Author: @joshuacords

LPS-118240 Not possible to set a "_score" sort in asc
https://issues.liferay.com/browse/LPS-118240

### CODE REVIEW

File: modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/sort/BaseSortTestCase.java [hunk 1]
```diff
@@ -120,4 +183,6 @@ protected void testDoubleFieldSortable(String fieldName) throws Exception {
 				fieldName, value));
 	}
 
+	protected String scoreParameter;
```
Comment (arboliveira) [inline modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/sort/BaseSortTestCase.java:186]: Delete field and introduce an abstract getter instead

File: modules/apps/portal-search-solr7/portal-search-solr7-impl/src/test/java/com/liferay/portal/search/solr7/internal/sort/SolrSortTest.java [hunk 2]
```diff
@@ -17,12 +17,21 @@
...
  * @author Wade Cao
  */
 public class SolrSortTest extends BaseSortTestCase {
 
+	@Before
+	@Override
+	public void setUp() throws Exception {
+		super.setUp();
+
+		scoreParameter = "score";
```
Comment (arboliveira) [inline modules/apps/portal-search-solr7/portal-search-solr7-impl/src/test/java/com/liferay/portal/search/solr7/internal/sort/SolrSortTest.java:32]: Do not assign; instead override an abstract getter

File: modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/sort/BaseSortTestCase.java [hunk 3]
```diff
@@ -34,6 +39,38 @@
  */
 public abstract class BaseSortTestCase extends BaseIndexingTestCase {
 
+	@Before
+	public void setUp() throws Exception {
```
Comment (arboliveira) [inline modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/sort/BaseSortTestCase.java:43]: Delete no-op setUp method

File: modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/sort/BaseSortTestCase.java [hunk 4]
```diff
@@ -34,6 +39,38 @@
...
+		stringValues.add("test value");
+		stringValues.add("test");
+
+		String fieldName = "testField";
+
+		addDocuments(
+			value -> DocumentCreationHelpers.singleText(fieldName, value),
+			stringValues.stream());
+
+		Sort[] sorts = {new Sort(scoreParameter, Sort.SCORE_TYPE, true)};
```
Comment (arboliveira) [inline modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/sort/BaseSortTestCase.java:60]: Invoke abstract getter

File: modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/sort/BaseSortTestCase.java [hunk 5]
```diff
@@ -34,6 +39,38 @@
...
+
+		assertKeywordSearchOrder(fieldName, sorts, stringValues);
+
+		sorts = new Sort[] {new Sort(scoreParameter, Sort.SCORE_TYPE, false)};
+
```
Comment (arboliveira) [inline modules/apps/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/sort/BaseSortTestCase.java:65]: Local variable should preferrably not be reassigned - recommend to inline value in usage point directly

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/sort/DefaultSortTranslator.java [hunk 6]
```diff
@@ -69,7 +69,12 @@ public void translate(
 
 			SortOrder sortOrder = SortOrder.ASC;
 
-			if (sort.isReverse() || sortFieldName.equals("_score")) {
+			String originalSortName = sort.getFieldName();
+
+			if (sort.isReverse() ||
+				(sortFieldName.equals("_score") &&
+				 !sortFieldName.equals(originalSortName))) {
```
Comment (arboliveira) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/sort/DefaultSortTranslator.java:76]: Remove this check - instead, add "_score" here as a similar no-op return:

https://github.com/liferay/liferay-portal/blob/2dde8068d881bf64213e055d12470148c9b5bf4b/modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/sort/DefaultSortTranslator.java#L128-L130

File: modules/apps/portal-search-elasticsearch6/portal-search-elasticsearch6-impl/src/main/java/com/liferay/portal/search/elasticsearch6/internal/sort/DefaultSortTranslator.java [hunk 7]
```diff
@@ -69,7 +69,12 @@ public void translate(
 
 			SortOrder sortOrder = SortOrder.ASC;
 
-			if (sort.isReverse() || sortFieldName.equals("_score")) {
+			String originalSortName = sort.getFieldName();
+
+			if (sort.isReverse() ||
+				(sortFieldName.equals("_score") &&
+				 !sortFieldName.equals(originalSortName))) {
```
Comment (arboliveira) [inline modules/apps/portal-search-elasticsearch6/portal-search-elasticsearch6-impl/src/main/java/com/liferay/portal/search/elasticsearch6/internal/sort/DefaultSortTranslator.java:76]: Remove this check - see comment at https://github.com/liferay-search/liferay-portal/pull/10/files#diff-0fd9f35f159ee8c153ccbb56758aeb2cR76

File: modules/apps/portal-search-solr7/portal-search-solr7-impl/src/main/java/com/liferay/portal/search/solr7/internal/search/engine/adapter/search/SearchSolrQueryAssemblerImpl.java [hunk 8]
```diff
@@ -253,7 +253,12 @@ protected void setSorts(
 
 			SolrQuery.ORDER order = SolrQuery.ORDER.asc;
 
-			if (sort.isReverse() || sortFieldName.equals("score")) {
+			String originalSortName = sort.getFieldName();
+
+			if (sort.isReverse() ||
+				(sortFieldName.equals("score") &&
+				 !sortFieldName.equals(originalSortName))) {
```
Comment (arboliveira) [inline modules/apps/portal-search-solr7/portal-search-solr7-impl/src/main/java/com/liferay/portal/search/solr7/internal/search/engine/adapter/search/SearchSolrQueryAssemblerImpl.java:260]: Remove this check - see comment at https://github.com/liferay-search/liferay-portal/pull/10/files#diff-0fd9f35f159ee8c153ccbb56758aeb2cR76

**Comment by joshuacords** [comment]
ci:test:sf:norebase

**Comment by joshuacords** [comment]
ci:test:search:norebase

**Comment by joshuacords** [comment]
ci:test:search-solr:norebase

**Comment by joshuacords** [comment]
ci:test:search-solr:norebase

**Comment by joshuacords** [comment]
ci:test:search:relevant

**Comment by joshuacords** [comment]
ci:test:search-solr:norebase

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
ci:test:relevant

**Comment by joshuacords** [comment]
ci:test:search

**Comment by joshuacords** [comment]
ci:test:search-solr

**Pull request closed without merging**